### ISSUE TITLE
Document attributes, in a document with a header, are not processed when using the include macro with a leveloffset=1

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor#1911]
# Given

.asciidoctorjs.adoc
```
= Asciidoctor.js
Guillaume Grossetie <guillaume@gmail.com>

:release-version: 1.5.5-3

== Install

[source,sh,subs=attributes+]
----
$ npm install asciidoctor@{release-version}
----
```
.asciidoctor-maven-plugin.adoc
```
= Asciidoctor Maven Plugin
:release-version: 1.5.5

== Install

[source,sh,subs=attributes+]
----
$ mvn install asciidoctor@{release-version}
----
```

.include.adoc
```
= Include

include::asciidoctorjs.adoc[leveloffset=1]

include::asciidoctor-maven-plugin.adoc[leveloffset=1]
```

# Then

![notworking](https://cloud.githubusercontent.com/assets/333276/20030388/24b8db16-a364-11e6-8726-3aa4b66f6352.png)

## Notes

If I add an empty line after the document header (ie. after the author line) then the document attributes are processed but in my opinion the author line should not be displayed:

![better](https://cloud.githubusercontent.com/assets/333276/20030396/4c57d898-a364-11e6-9fe9-68d5c34f4500.png)

`:noheader:` attribute does not help (reference: http://asciidoctor.org/docs/user-manual/#doc-header)

Also the result is the same when I'm rendering `asciidoctorjs.adoc` with or without an empty line after the document header:

![resultsame](https://cloud.githubusercontent.com/assets/333276/20030407/9708ae44-a364-11e6-93d6-3a8252a54520.png)

So I don't know if an empty line is mandatory after the document header ? And is it possible to not render the document header of an included document ?

**Comment by mojavelinux** [comment]
Yes, this is a known limitation. The include macro doesn't really recognize document structure, so you really have to look at the concatenated document to see what the parser sees.

I faced this problem in the epub3 converter, which requires that chapters be added to the spine document using includes. Of course, I still wanted the header to be parsed...so I wrote an include processor that handles the include in a special way.

See https://github.com/asciidoctor/asciidoctor-epub3/blob/master/lib/asciidoctor-epub3/spine_item_processor.rb

It turned out to be a bit more complicated than I expected.

I think this really gets us back to the fact that we need a subdoc block macro. We need to be able to stitch two standalone documents together in such a way that honors the header, honors the location, and so forth. We keep coming back to this. We are really abusing the include directive by using it to weave documents (not just partials) together.

...but I'm open to other ideas as well.
